src/core/server.cc

```diff
@@ -149,9 +149,10 @@ InferenceServer::Init()
   CudaMemoryManager::Options cuda_options(
       min_supported_compute_capability_, cuda_memory_pool_size_);
   status = CudaMemoryManager::Create(cuda_options);
+  // If CUDA memory manager can't be created, just log error as the
+  // server can still function properly
   if (!status.IsOk()) {
-    ready_state_ = ServerReadyState::SERVER_FAILED_TO_INITIALIZE;
-    return status;
+    LOG_ERROR << status.Message();
```

Comment:
Don't you need to do something special to fail appropriately if load/unload is called and the manager didn't create correctly?

Reply:
The Alloc / Free functions are checking if the CUDA memory manager is created, if not, the functions will return error